tdms/src/argument_parser.cpp

```diff
@@ -0,0 +1,120 @@
+#include "stdexcept"
+#include "argument_parser.h"
+
+using namespace std;
+
+
+ArgumentParser::ArgumentParser() = default;
+
+ArgumentNamespace ArgumentParser::parse_args(int nargs, char *argv[]) {
+
+  auto args = ArgumentNamespace(nargs, argv);
+
+  if (args.have_flag("-h")){
+    print_help_message();
+    exit(0);
+  }
+
+  if( (args.num_non_flag != 2) &&  (args.num_non_flag != 3) ){
```

Comment:
```suggestion
  if !((args.num_non_flag == 2) || (args.num_non_flag == 3)){
```
I always struggle to get my head around > 1 not statements combined with another operator, and think this is the clearer way to say "If number of args isn't 2 or 3"

Reply:
Thanks for pointing that out, although I actually find that equally difficult to parse. Will create a method 👌🏼